builtin.c

```diff
@@ -489,35 +489,93 @@ static OP *ck_builtin_funcN(pTHX_ OP *entersubop, GV *namegv, SV *ckobj)
-    { "builtin::ceil",       &XS_builtin_func1_scalar, &ck_builtin_func1, OP_CEIL,       false },
-    { "builtin::floor",      &XS_builtin_func1_scalar, &ck_builtin_func1, OP_FLOOR,      false },
-    { "builtin::is_tainted", &XS_builtin_func1_scalar, &ck_builtin_func1, OP_IS_TAINTED, false },
-    { "builtin::trim",       &XS_builtin_trim,         &ck_builtin_func1, 0,             false },
-    { "builtin::stringify",  &XS_builtin_func1_scalar, &ck_builtin_func1, OP_STRINGIFY,  true },
-
-    { "builtin::created_as_string", &XS_builtin_created_as_string, &ck_builtin_func1, 0, true },
-    { "builtin::created_as_number", &XS_builtin_created_as_number, &ck_builtin_func1, 0, true },
+    { "is_bool",   NO_BUNDLE, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_IS_BOOL,    true  },
+    { "weaken",         5039, &XS_builtin_func1_void,   &ck_builtin_func1, OP_WEAKEN,     false },
+    { "unweaken",       5039, &XS_builtin_func1_void,   &ck_builtin_func1, OP_UNWEAKEN,   false },
+    { "is_weak",        5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_IS_WEAK,    false },
+    { "blessed",        5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_BLESSED,    false },
+    { "refaddr",        5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_REFADDR,    false },
+    { "reftype",        5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_REFTYPE,    false },
+    { "ceil",           5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_CEIL,       false },
+    { "floor",          5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_FLOOR,      false },
+    { "is_tainted",     5039, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_IS_TAINTED, false },
+    { "trim",           5039, &XS_builtin_trim,         &ck_builtin_func1, 0,             false },
+    { "stringify", NO_BUNDLE, &XS_builtin_func1_scalar, &ck_builtin_func1, OP_STRINGIFY,  true },
+
+    { "created_as_string", NO_BUNDLE, &XS_builtin_created_as_string, &ck_builtin_func1, 0, true },
+    { "created_as_number", NO_BUNDLE, &XS_builtin_created_as_number, &ck_builtin_func1, 0, true },
 
     /* list functions */
-    { "builtin::indexed",          &XS_builtin_indexed,          &ck_builtin_funcN, 0, false },
-    { "builtin::export_lexically", &XS_builtin_export_lexically, NULL,              0, true },
+    { "indexed",               5039, &XS_builtin_indexed,          &ck_builtin_funcN, 0, false },
+    { "export_lexically", NO_BUNDLE, &XS_builtin_export_lexically, NULL,              0, true },
 
-    { NULL, NULL, NULL, 0, false }
+    { NULL, 0, NULL, NULL, 0, false }
 };
 
+static bool S_parse_version(const char *vstr, int *vmajor, int *vminor)
+{
+    /* Parse a string like "5.35" to yield 5 and 35. Ignores an optional
+     * trailing third component e.g. "5.35.7". Returns false on parse errors.
+     */
+
+    size_t len;
+
+    if(sscanf(vstr, "%d.%d%zn", vmajor, vminor, &len) < 2)
```

Comment:
Since you know you only want 3 trailing digits max, can the `sscanf` patterns be `%d.%3d%n` and `%3d%n`? That would hopefully allow simplifying to something like this (not tested):

```
static bool S_parse_version(const char *vstr, int *vmajor, int *vminor)
{
    /* Parse a string like "5.35" to yield 5 and 35. Ignores an optional
     * trailing third component e.g. "5.35.7". Returns false on parse errors.
     */

    size_t len;

    if(sscanf(vstr, "%d.%d%zn", vmajor, vminor, &len) < 2)
        return FALSE;

    vstr += len;

    if (!vstr[0])
        return TRUE;
    
    if(vstr[0] == '.') {
        vstr++;

        int _dummy;
        if(sscanf(vstr, "%3d%zn", &_dummy, &len) < 1)
            return FALSE;
        vstr += len;
    }

    if(vstr[0])
        return FALSE;

    return TRUE;
}
```